source/extensions/filters/http/stateful_session/stateful_session.cc

```diff
@@ -0,0 +1,75 @@
+#include "source/extensions/filters/http/stateful_session/stateful_session.h"
+
+#include <memory>
+
+#include "source/common/config/utility.h"
+#include "source/common/http/utility.h"
+#include "source/common/upstream/load_balancer_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace StatefulSession {
+
+StatefulSessionConfig::StatefulSessionConfig(const ProtoConfig& config,
+                                             Server::Configuration::CommonFactoryContext& context) {
+  std::vector<envoy::config::core::v3::HealthStatus> host_statuses;
+  for (int i = 0; i < config.host_statuses_size(); i++) {
+    host_statuses.push_back(config.host_statuses(i));
+  }
+  host_statuses_ = Upstream::LoadBalancerContextBase::createOverrideHostStatus(host_statuses);
+
+  auto& factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<Envoy::Http::SessionStateFactoryConfig>(
+          config.session_state().name());
+
+  factory_ = factory.createSessionStateFactory(config.session_state().typed_config(), context);
+}
+
+PerRouteStatefulSession::PerRouteStatefulSession(
+    const PerRouteProtoConfig& config, Server::Configuration::CommonFactoryContext& context) {
+  if (config.override_case() == PerRouteProtoConfig::kDisabled) {
+    disabled_ = true;
+    return;
```

Comment:
Is there a use-case for someone providing the route-specific config with disabled = false and expecting the overall config to take effect?

Reply:
🤔  I think there should be no such use-case. If overall config is wanted, then the user just need not provide route-specific config.